packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/include/video_player_avfoundation/FVPVideoPlayer.h

```diff
@@ -0,0 +1,109 @@
+@property(nonatomic) CGAffineTransform preferredTransform;
+/// Indicates whether the video player has been disposed.
+@property(nonatomic, readonly) BOOL disposed;
+/// Indicates whether the video player is currently playing.
+@property(nonatomic, readonly) BOOL isPlaying;
+/// Indicates whether the video player is set to loop.
+@property(nonatomic) BOOL isLooping;
+/// Indicates whether the video player has been initialized.
+@property(nonatomic, readonly) BOOL isInitialized;
+/// The updater that drives callbacks to the engine to indicate that a new frame is ready.
+@property(nonatomic, nullable) FVPFrameUpdater *frameUpdater;
+/// The display link that drives frameUpdater.
+@property(nonatomic, nullable) FVPDisplayLink *displayLink;
+/// Whether a new frame needs to be provided to the engine regardless of the current play/pause
+/// state (e.g., after a seek while paused). If YES, the display link should continue to run until
+/// the next frame is successfully provided.
+@property(nonatomic, assign) BOOL waitingForFrame;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// Initializes a new instance of FVPVideoPlayer with the given URL, frame updater, display link,
+/// HTTP headers, AV factory, and registrar.
+- (instancetype)initWithURL:(NSURL *)url
+               frameUpdater:(FVPFrameUpdater *)frameUpdater
+                displayLink:(FVPDisplayLink *)displayLink
+                httpHeaders:(nonnull NSDictionary<NSString *, NSString *> *)headers
+                  avFactory:(id<FVPAVFactory>)avFactory
+                  registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
+
+/// Initializes a new instance of FVPVideoPlayer with the given AVPlayerItem, frame updater, display
+/// link, AV factory, and registrar.
+- (instancetype)initWithPlayerItem:(AVPlayerItem *)item
+                      frameUpdater:(FVPFrameUpdater *)frameUpdater
+                       displayLink:(FVPDisplayLink *)displayLink
+                         avFactory:(id<FVPAVFactory>)avFactory
+                         registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
+
+/// Initializes a new instance of FVPVideoPlayer with the given asset, frame updater, display link,
+/// AV factory, and registrar.
+- (instancetype)initWithAsset:(NSString *)asset
+                 frameUpdater:(FVPFrameUpdater *)frameUpdater
+                  displayLink:(FVPDisplayLink *)displayLink
+                    avFactory:(id<FVPAVFactory>)avFactory
+                    registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
+
+NS_ASSUME_NONNULL_END
+
+/// Updates the playing state of the video player.
+- (void)updatePlayingState;
+
+/// Disposes the video player and releases any resources it holds.
+- (void)dispose;
+
+/// Disposes the video player without touching the event channel. This
+/// is useful for the case where the Engine is in the process of deconstruction
+/// so the channel is going to die or is already dead.
+- (void)disposeSansEventChannel;
```

Comment:
Why does this need to be public?

Reply:
It is referenced in `FVPVideoPlayerPlugin.m`

https://github.com/flutter/packages/blob/6c7a61d1fc30e28edd7851ce6c558fafa0f82603/packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m#L711-L715